kinesis/kinesis.go

```diff
@@ -331,7 +425,7 @@ func (outputPlugin *OutputPlugin) randomString() string {
 
 // getPartitionKey returns the value for a given valid key
 // if the given key is emapty or invalid, it returns a random string
-func (outputPlugin *OutputPlugin) getPartitionKey(record map[interface{}]interface{}) string {
+func (outputPlugin *OutputPlugin) getPartitionKey(records []*kinesis.PutRecordsRequestEntry, record map[interface{}]interface{}) string {
```

Comment:
Why do we need this new parameter `records`?

Reply:
This function uses the record list to set the index. Previously, the record slice was stored in the plugin structure. That does not work if the plugin is concurrent though- each goroutine needs its own slice to store records.

Reply:
Maybe I am wrong. But, I can see, you already removed the line(349) where we used `records`. I can't see the usage of your new `records` parameter.

Reply:
Yes, I'll remove the records parameter.  The input record is the only require parameter now.